test/integ/basic.cpp

```diff
@@ -627,6 +645,30 @@ LT_BEGIN_AUTO_TEST(basic_suite, postprocessor)
     curl_easy_cleanup(curl);
 LT_END_AUTO_TEST(postprocessor)
 
+LT_BEGIN_AUTO_TEST(basic_suite, same_key_different_value)
+    arg_value_resource resource;
+    ws->register_resource("base", &resource);
+    curl_global_init(CURL_GLOBAL_ALL);
+    string s;
+    CURL *curl = curl_easy_init();
+    CURLcode res;
+    curl_easy_setopt(curl, CURLOPT_URL, "localhost:8080/base?arg=beep&arg=boop&arg=hello&arg=what");
+    curl_easy_setopt(curl, CURLOPT_POSTFIELDS, "arg=beep&arg=boop&arg=hello&arg=what");
+    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writefunc);
+    curl_easy_setopt(curl, CURLOPT_WRITEDATA, &s);
+    struct curl_slist *list = NULL;
+    // Change the content type to trigger "normal" POST processing,
+    // otherwise the file processing logic is triggered which does
+    // not set the multiple arg values as expected.
```

Comment:
I think you might have lost me here. Why does this need to be done?

Reply:
Without a different content type the defaults that cURL sets trigger the file processing code, but in this test I am just trying to check the "normal" arg handling with multiple values.